File: ClaimPortalJava/src/main/java/com/mt/claimportaljava/services/ClaimService.java
package com.mt.claimportaljava.services;

import com.mt.claimportaljava.dtos.CreateClaimDto;
import com.mt.claimportaljava.models.Claim;
import com.mt.claimportaljava.repositories.ClaimRepository;
import com.mt.claimportaljava.utils.model.ClaimBuilder;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Date;
import java.util.Map;

@Service
public class ClaimService {
    private ClaimRepository claimRepository;
    private ClaimBuilder claimBuilder;

    @Autowired
    public ClaimService(ClaimRepository claimRepository) {
        this.claimRepository = claimRepository;
    }

    @Autowired
    public void setClaimBuilder(ClaimBuilder claimBuilder) {
        this.claimBuilder = claimBuilder;
    }

    public Claim getClaimByClaimNumber(String claimNumber) {
        return claimRepository.findClaimByClaimNumber(claimNumber);
    }

    public String createClaim(CreateClaimDto createClaimDto) {
        claimBuilder.withDataFromDto(createClaimDto);
        Claim claim = claimBuilder.build();
        claimRepository.save(claim);
        return claim.getClaimNumber();
    }

    public void updateClaim(String claimNumber, Map<String, Object> fields) {
        Claim claim = claimRepository.findClaimByClaimNumber(claimNumber);
        if (fields.containsKey("description")) {
            claim.setDescription((String) fields.get("description"));
        }
        if (fields.containsKey("lossDate")) {
            claim.setLossDate((Date) fields.get("lossDate"));
        }
        claimRepository.save(claim);
    }

    @Transactional
    public void deleteClaim(String claimNumber) {
        claimRepository.deleteByClaimNumber(claimNumber);
    }

}

File: ClaimSystemPython/src/services/claim_service.py
from sqlalchemy import select
from sqlalchemy.orm import Session

from src.models.claim import Claim
from src.utils.database import DB
from src.utils.model.claim_builder import build_claim, build_claim_response


def get_claim_by_claim_number(claim_number):
    with Session(DB.get_instance().engine) as session:
        query = select(Claim).where(Claim.claim_number == claim_number)
        claim = session.execute(query).scalar()
    return build_claim_response(claim)


def create_claim(create_claim_schema):
    with Session(DB.get_instance().engine) as session:
        claim = build_claim(create_claim_schema)

        session.add(claim)
        session.flush()
        session.refresh(claim)
        claim_number = claim.claim_number
        session.commit()

    return claim_number


def update_claim(claim_number, fields):
    with Session(DB.get_instance().engine) as session:
        query = select(Claim).where(Claim.claim_number == claim_number)
        claim = session.execute(query).scalar()

        if "description" in fields:
            claim.description = fields["description"]
        if "lossDate" in fields:
            claim.loss_date = fields["lossDate"]

        session.add(claim)
        session.commit()


def delete_claim(claim_number):
    with Session(DB.get_instance().engine) as session:
        query = select(Claim).where(Claim.claim_number == claim_number)
        claim = session.execute(query).scalar()

        session.delete(claim)
        session.commit()
